knife/lib/chef/knife/bootstrap.rb

```diff
@@ -950,12 +950,25 @@ def ssh_opts
         opts = {}
         return opts if winrm?
 
-        opts[:non_interactive] = true # Prevent password prompts from underlying net/ssh
+        opts[:non_interactive] = ssh_config_tty ? false : true # Prevent password prompts from underlying net/ssh
         opts[:forward_agent] = (config[:ssh_forward_agent] === true)
         opts[:connection_timeout] = session_timeout
         opts
       end
 
+      # checks the ~/.ssh/config file to check RequestTTY is set to yes
+      def ssh_config_tty
+        if File.exists?(File.join(Dir.home, ".ssh/config"))
+          ssh_config_file = File.open(File.join(Dir.home, ".ssh/config")).read
+          if ssh_config_file.include?("RequestTTY")
+            tty_key_index = ssh_config_file.index("RequestTTY")
+            tty_value = ssh_config_file[tty_key_index...].split[1].downcase
+            return true if tty_value == "yes"
+          end
+        end
+        false
+      end
```

Comment:
how do you know that `RequestTTY` is being set for that host? What if I have:

```
Host mycoolhost
  RequestTTY
```  

but am trying to bootstrap `adifferenthostname`? You'll return true, but incorrectly.

This doesn't seem like the right approach. I would:

1. File an upstream bug against Net::SSH to parse `RequestTTY` properly
2. As a work around, you could add a command-line option to knife, or possibly we might accept checking an environment variable (someone should confirm this would be an OK thing to do, I don't work for progress) ... something like `CHEF_SSH_REQUEST_TTY=true` or some-such.

Reply:
Yeah make sense RequestTTY is host specific. Adding a command line option to knife would be a better work around it seems as of now.